### PR TITLE
Fix: Kit-Library - Connect Error caused by URL changes in Kit and template Library [ED-4353]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: core/common/modules/connect/apps/base-app.php [hunk 1]
```diff
@@ -341,11 +341,10 @@ protected function add_notice( $content, $type = 'success' ) {
 	}
 
 	/**
-	 * @deprecated Please use `http_request` method instead of this method.
-	 *
 	 * @param       $action
 	 * @param array $request_body
 	 * @param false $as_array
+	 * @param array $options
```
Comment (matipojo) [inline core/common/modules/connect/apps/base-app.php:347]: Remove

File: core/common/modules/connect/apps/base-app.php [hunk 2]
```diff
@@ -720,12 +721,14 @@ private function get_generated_urls( $endpoint ) {
...
 	}
 
-	/**
-	 * @since 2.3.0
-	 * @access public
-	 */
-	public function __construct() {
-		add_action( 'admin_notices', [ $this, 'admin_notice' ] );
+	private function init_auth_mode() {
+		$is_rest = defined( 'REST_REQUEST' ) && REST_REQUEST;
```
Comment (matipojo) [inline core/common/modules/connect/apps/base-app.php:725]: Test

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/common/modules/connect/apps/base-app.php (hunk 1 above)
Comment (Nevoss) [reply core/common/modules/connect/apps/base-app.php]: Done

File: core/common/modules/connect/apps/base-app.php (hunk 2 above)
Comment (Nevoss) [reply core/common/modules/connect/apps/base-app.php:724]: Done

**Review by YitzhakAvm — APPROVED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 3]
```diff
@@ -78,11 +78,50 @@ public function test_http_request() {
...
 		// Arrange
+		$user = $this->act_as_admin();
+
+		$filter = function () {
+			return true;
+		};
+
+		update_user_option( $user->ID, $this->app_stub->get_option_name(), [ 'a' => 1 ] );
+
+		add_filter( 'wp_doing_ajax', $filter );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:91]:
```suggestion
		add_filter( 'wp_doing_ajax', '__return_true' );
```

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 4]
```diff
@@ -78,11 +78,50 @@ public function test_http_request() {
...
 	}
 
-	public function test_http_request__return_error_when_not_response_code() {
+	public function test_http_request__unauthorized_request_when_auth_mode_xhr() {
 		// Arrange
+		$user = $this->act_as_admin();
+
+		$filter = function () {
+			return true;
+		};
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:87]:
```suggestion
```

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php [hunk 5]
```diff
@@ -78,11 +78,50 @@ public function test_http_request() {
...
+
+		update_user_option( $user->ID, $this->app_stub->get_option_name(), [ 'a' => 1 ] );
+
+		add_filter( 'wp_doing_ajax', $filter );
+
+		$this->app_stub = new Mock_App();
+		$this->app_stub->set_http( $this->http_stub );
+
+		remove_filter( 'wp_doing_ajax', $filter );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:96]:
```suggestion
		remove_filter( 'wp_doing_ajax', '__return_true' );
```

**Review by matipojo — CHANGES_REQUESTED** [review]

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 3 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:91]: Done

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 4 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:87]: Done

File: tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php (hunk 5 above)
Comment (Nevoss) [reply tests/phpunit/elementor/core/common/modules/connect/apps/test-common-app.php:96]: Doen

**Review by matipojo — APPROVED** [review]

**Pull request merged**